src/System.Buffers.Experimental/System/Buffers/BufferExtensions.cs

```diff
@@ -117,5 +118,107 @@ internal static int IndexOfStraddling(this ReadOnlySpan<byte> first, IReadOnlyMe
 
             return -1;
         }
+
+        static readonly int s_longSize = Vector<ulong>.Count;
+        static readonly int s_byteSize = Vector<byte>.Count;
+
+        public static int IndexOfVectorized(this Span<byte> buffer, byte value)
+        {
+            Debug.Assert(s_longSize == 4 || s_longSize == 2);
+
+            var byteSize = s_byteSize;
+
+            if (buffer.Length < byteSize * 2 || !Vector.IsHardwareAccelerated) return buffer.IndexOf(value);
+
+            Vector<byte> match = new Vector<byte>(value);
```

Comment:
Needs jit fix https://github.com/dotnet/coreclr/pull/7683 to be performant; else use 
```csharp
Vector<byte> match = Vector.AsVectorByte(new Vector<uint>(value * 0x01010101u));
```